### PR TITLE
Add global registry from docker config file on woodpecker server

### PR DESCRIPTION
**Pull request by imduffy15**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,8 +10,10 @@ require (
 	github.com/cpuguy83/go-md2man/v2 v2.0.0 // indirect
 	github.com/dgrijalva/jwt-go v0.0.0-20150904212456-c1da56349675
 	github.com/dimfeld/httptreemux v5.0.1+incompatible
+	github.com/docker/cli v0.0.0-20200303215952-eb310fca4956
 	github.com/docker/distribution v2.7.1+incompatible
-	github.com/docker/docker v17.12.0-ce-rc1.0.20200309214505-aa6a9891b09c+incompatible
+	github.com/docker/docker v1.13.1
```
Comment (laszlocph) [inline go.mod:15]: Can you rebase to latest master? I merged your dep upgrade PR and seems these changes are reverting some of those back.

File: plugins/registry/combine.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+)
+
+type combined struct {
+	registries []model.ReadOnlyRegistryService
+	dbRegistry model.RegistryService
+}
+
+func Combined(dbRegistry model.RegistryService, registries ...model.ReadOnlyRegistryService) model.RegistryService {
+	registries = append(registries, dbRegistry)
+	return &combined{
```
Comment (laszlocph) [inline plugins/registry/combine.go:14]: This is great

File: plugins/registry/combine.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		registries: registries,
+		dbRegistry: dbRegistry,
+	}
+}
+
+func (c combined) RegistryFind(repo *model.Repo, name string) (*model.Registry, error) {
+	for _, registry := range c.registries {
```
Comment (laszlocph) [inline plugins/registry/combine.go:21]: Looks as if only the filesystem registries are looped through. The call should be delegated to `dbRegistry` too, shouldn't it?

File: plugins/registry/combine.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+			return nil, err
+		}
+		if res != nil {
+			return res, nil
+		}
+	}
+	return nil, nil
+}
+
+func (c combined) RegistryList(repo *model.Repo) ([]*model.Registry, error) {
```
Comment (laszlocph) [inline plugins/registry/combine.go:33]: Few typos in this one: registory, registeries

File: plugins/registry/combine.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	var registeries []*model.Registry
+	for _, registory := range c.registries {
```
Comment (laszlocph) [inline plugins/registry/combine.go:35]: Shouldn't the call be propagated to `dbRegistries` too?

File: go.mod (hunk 1 above)
Comment (imduffy15) [reply go.mod]: Fixed, had rebased but the docker cli dependency changed things here

File: plugins/registry/combine.go (hunk 3 above)
Comment (imduffy15) [reply plugins/registry/combine.go:21]: Shouldn't have to, dbRegistry is added to the list https://github.com/laszlocph/woodpecker/pull/120/files#diff-60ca29804ed147e90820993d6f198f1fR13

File: plugins/registry/combine.go (hunk 5 above)
Comment (imduffy15) [reply plugins/registry/combine.go]: Shouldn't have to, dbRegistry is added to the list https://github.com/laszlocph/woodpecker/pull/120/files#diff-60ca29804ed147e90820993d6f198f1fR13

File: plugins/registry/combine.go (hunk 3 above)
Comment (laszlocph) [reply plugins/registry/combine.go:21]: My bad. Thanks for clarification.

**Pull request merged**